### DISCUSSION TITLE
macros in completions?

### DISCUSSION THREAD

**Discussion by johnnywz00** [toerob/vscode-tads3tools#13] [Ideas]
Would it be a major task for the completions to include user-defined macros? Also, to allow peek/goto definition from a macro (maybe provided that the macro stands for a single symbol?)

**Comment by toerob** [comment]
Good idea!

**Comment by johnnywz00** [comment]
Just to clarify, I use a variety of different macro types. Not all of them would really apply to the peek definition (because they're simplifications of repetitive code blocks), but some would. Here are three types that could benefit from participating in peek definition:
1. The macro is simply an abbreviation for an actual symbol, used in either a calling or a defining context. Examples:
#define mI    moveInto
#define sD   specialDesc
2. The macro is not a simple symbol, but may evaluate to a single symbol. Examples:
#define gpc    libGlobal.playerChar
#define gor   gpc.getOutermostRoom
3. The macro is a single symbol but with a parameter list, used in a defining/overriding context. (Yes, I just have to remember the parameter names I'm using). Examples:
#define cORC   canObjReachContents(obj)
#define slpW   showListPrefixWide(itemCount,pov,parent)
